src/components/study-container.js

```diff
@@ -382,36 +385,70 @@ export function StudyContainer({ view, onChangeTab }) {
-        fetchPath(studyUuid)
-            .then((response) => {
-                const parents = response
-                    .slice(1)
-                    .map((parent) => parent.elementName);
-
-                const studyName = response[0]?.elementName;
-                const path = computeFullPath(parents);
-                setStudyName(studyName);
-                setStudyPath(path);
-
-                document.title = computePageTitle(
-                    initialTitle,
-                    studyName,
-                    parents
-                );
-            })
-            .catch((errorMessage) => {
-                document.title = initialTitle;
-                snackError(errorMessage, 'LoadStudyAndParentsInfoError');
-            });
-    }, [studyUuid, initialTitle, snackError]);
+    const fetchStudyPath = useCallback(
+        (studyUuid) => {
+            fetchPath(studyUuid)
+                .then((response) => {
+                    const parents = response
+                        .slice(1)
+                        .map((parent) => parent.elementName);
+
+                    const studyName = response[0]?.elementName;
+                    const path = computeFullPath(parents);
+                    setStudyName(studyName);
+                    setStudyPath(path);
+
+                    document.title = computePageTitle(
+                        initialTitle,
+                        studyName,
+                        parents
+                    );
+                })
+                .catch((errorMessage) => {
+                    document.title = initialTitle;
+                    snackError(errorMessage, 'LoadStudyAndParentsInfoError');
+                });
+        },
+        [initialTitle, snackError]
+    );
+
+    useEffect(() => {
+        // create ws at mount event
+        wsRef.current = connectNotificationsWsUpdateDirectories();
+
+        wsRef.current.onmessage = function (event) {
+            const eventData = JSON.parse(event.data);
+            if (eventData.headers) {
+                if (
+                    eventData.headers['notificationType'] ===
+                    directoriesNotificationType.UPDATE_DIRECTORY
+                ) {
+                    fetchStudyPath(studyUuid);
+                }
+            }
+        };
+
+        wsRef.current.onclose = function () {
+            console.error('Unexpected Notification WebSocket closed');
+        };
+        wsRef.current.onerror = function (event) {
+            console.error('Unexpected Notification WebSocket error', event);
+        };
+        // We must save wsRef.current in a variable to make sure that when close is called it refers to the same instance.
+        // That's because wsRef.current could be modify outside of this scope.
+        const wsToClose = wsRef.current;
+        // cleanup at unmount event
+        return () => {
+            wsToClose.close();
+        };
+    }, [fetchStudyPath, studyUuid]);
```

Comment:
we want the websocket creation at mount event then we must have 0 dependencies here for this effect.

Reply:
it's fixed